academy/controllers/main.py

```diff
@@ -3,12 +3,21 @@
 
 class Academy(http.Controller):
 
-    @http.route('/academy/academy/', auth='public')
+    @http.route('/academy/academy/', auth='public', website=True)
     def index(self, **kw):
+        Teachers = http.request.env['academy.teachers']
         return http.request.render('academy.index', {
-            'teachers': [
-                'pepe rodriguez',
-                'juan',
-                'pablo',
-            ]
-        })
\ No newline at end of file
+            'teachers': Teachers.search([])
+        })
+
+    @http.route('/academy/<name>', auth='public', website=True)
+    def teacher(self, name):
+        return '<h1>%s</h1>' % name
+    
+    @http.route('/academy/<int:id>', auth='public', website=True)
+    def int_teacher(self, id):
+        return '<h1>{} ({})</h1>'.format(id, type(id).__name__)
```

Comment:
Use concatenation by % instead

```suggestion
        return '<h1>%s (%s)</h1>' % (id, type(id).__name__)
```

I also don't get why we are returning the id and the type of the id

Reply:
The purpose of this controller is to show that controllers can validate data